### PR TITLE
[SYCL] Add missing double support in cross()

### PR DESCRIPTION
**Pull request by nawrinsu**
This patch adds missing double3 and double4 support in
geometric function cross(). Currently, it only supports
float types.

Signed-off-by: Nawrin Sultana <nawrin.sultana@intel.com>

### CODE REVIEW

**Review by s-kanaev — COMMENTED** [review]
The change looks good.
Though, the test doesn't pass on CUDA.

File: sycl/test/basic_tests/vector_cross.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %RUN_ON_HOST %t.out
```
Comment (s-kanaev) [inline sycl/test/basic_tests/vector_cross.cpp:2]: Is there any motivation to limit testing to host only?

**Comment by nawrinsu** [comment]
PR for test in llvm-test-suite: https://github.com/intel/llvm-test-suite/pull/114

File: sycl/test/basic_tests/vector_cross.cpp (hunk 1 above)
Comment (romanovvlad) [reply sycl/test/basic_tests/vector_cross.cpp:2]: I guess, the reason is to avoid having dependency on runtime in the in-tree test.

Comment (s-kanaev) [reply sycl/test/basic_tests/vector_cross.cpp:2]: Sounds reasonable.

**Review by s-kanaev — APPROVED** [review]
LGTM.

**Pull request merged**